pymodbus/framer/socket_framer.py

```diff
@@ -36,13 +37,32 @@ class ModbusSocketFramer(ModbusFramer):
 
     method = "socket"
 
-    def __init__(self, decoder, client=None):
+    def __init__(self, decoder, **kwargs):
         """Initialize a new instance of the framer.
 
         :param decoder: The decoder factory implementation to use
         """
-        super().__init__(decoder, client)
+        super().__init__(decoder, **kwargs)
         self._hsize = 0x07
+        self.transport = kwargs.get('transport')
+
+    def _validate_slave_id(self, slaves: list, single: bool) -> bool:
+        """Validate if the received data is valid for the client.
```

Comment:
slaves is a list of int, not just any list.

Reply:
Just copied the signature from [ModbusFramer](https://github.com/pymodbus-dev/pymodbus/blob/131d30391efd9c64f1fec857ae0372746374e6d2/pymodbus/framer/base.py#L48), didn't want to change the signture, but yes it can be more explicit, because with this pull request it is not `int` anymore.
I would like to change to:
```python
def _validate_slave_id(self, slaves: Sequence[Union[int, Tuple[str, int]]], single: bool) -> bool:
```
, but because ModbusFramer is requesting a list, we need to change ModbusFramer or use list instead of sequence.
`int` is still possible, because of  `0` used for the single option (see [ModbusServerContext](https://github.com/pymodbus-dev/pymodbus/blob/131d30391efd9c64f1fec857ae0372746374e6d2/pymodbus/datastore/context.py#L133)) and if the transport is not capable of delivering the peer information, the uid is still used.

Reply:
This is the beauty of mypy, you need to think carefully about how the parameters are used....but the advantage is that it get clearly documented.